### DISCUSSION TITLE
Interpolation Trait System

### DISCUSSION THREAD

**Discussion by termi-official** [Ferrite-FEM/Ferrite.jl#863] [Ideas]
With https://github.com/Ferrite-FEM/Ferrite.jl/pull/798 we are working our way towards more complex interpolation spaces (thanks again for working on this Knut!).

However, these function spaces break some internal assumptions and the come with some naming issues, e.g. Hcurl and Hdiv spaces are technically discontinuous, but are treated in a different way than L2 spaces, for which we introduced the "is_discontinuous" helpers.

My idea to handle this in an extensible way is to introduce a trait system. The 3 most important traits which I see for now are

1. conformity type (e.g. H1,Hcurl,Hdiv,L2 with https://github.com/Ferrite-FEM/Ferrite.jl/pull/798 which completes De-Rahm space support and more complicates spaces will follow)
2. interpolation kind, i.e. nodal or modal functions
3. geometry mapping type, e.g. identity, covariant Piola, contravariant Piola, ...

We could add more traits on demand as we start to add support for more interpolation spaces.

**Comment by KnutAM** [comment]
Would be useful for #867 which currently (but wrong after 798) uses `is_discontinuous`.
